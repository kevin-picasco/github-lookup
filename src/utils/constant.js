export const GITHUB_API_URL = 'https://api.github.com';
export const GITHUB_API_SORT_DIRECTION_ASC = 'asc';
export const GITHUB_API_SORT_DIRECTION_DESC = 'desc';