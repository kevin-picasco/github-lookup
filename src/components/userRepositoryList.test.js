import React from 'react';
import format from 'string-format';
import { Provider } from 'react-redux';
import { mount, shallow } from 'enzyme';
import configureStore from "redux-mock-store";
import * as Constant from '../utils/constant';
import UserRepositoryList from "./userRepositoryList";

describe('userRepositoryList', () => {
    it('render <UserRepositoryList> correctly when loading is in progress', () => {
        const mockStore = configureStore();

        const mockData = mockStore({
            userRepositoryReducer: {
                data: null,
                error: null,
                isLoading: true
            }
        });

        const component = shallow(
            <Provider store={mockData}>
                <UserRepositoryList />
            </Provider>);

        expect(component).toMatchSnapshot();
    });

    it('render <UserRepositoryList> correctly when there is an error', () => {
        const mockStore = configureStore();

        const mockData = mockStore({
            userRepositoryReducer: {
                data: null,
                error: format(Constant.MESSAGE_NOT_FOUND, 'User'),
                isLoading: false
            }
        });

        const component = shallow(
            <Provider store={mockData}>
                <UserRepositoryList />
            </Provider>);

        expect(component).toMatchSnapshot();
    });

    it('render <UserRepositoryList> correctly when userRepositories are empty', () => {
        const mockStore = configureStore();

        const mockData = mockStore({
            userRepositoryReducer: {
                data: [],
                error: null,
                isLoading: false
            }
        });

        const component = shallow(
            <Provider store={mockData}>
                <UserRepositoryList />
            </Provider>);

        expect(component).toMatchSnapshot();
    });

    it('render <UserRepositoryList> correctly when userRepositories have data', () => {
        const mockStore = configureStore();

        const mockData = mockStore({
            userRepositoryReducer: {
                data: [
                    {
                        id: 218454167,
                        name: 'github-lookup',
                        created_at: '2019-10-30T06:01:25Z'
                    }
                ],
                error: null,
                isLoading: false
            }
        });

        const component = mount(
            <Provider store={mockData}>
                <UserRepositoryList />
            </Provider>);

        expect(component).toMatchSnapshot();
    });
})