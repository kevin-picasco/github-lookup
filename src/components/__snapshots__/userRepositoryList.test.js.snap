// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`userRepositoryList render <UserRepositoryList> correctly when loading is in progress 1`] = `ShallowWrapper {}`;

exports[`userRepositoryList render <UserRepositoryList> correctly when there is an error 1`] = `ShallowWrapper {}`;

exports[`userRepositoryList render <UserRepositoryList> correctly when userRepositories are empty 1`] = `ShallowWrapper {}`;

exports[`userRepositoryList render <UserRepositoryList> correctly when userRepositories have data 1`] = `ReactWrapper {}`;
